osquery/tables/events/windows/windows_events.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <boost/property_tree/ptree.hpp>
+
+#include <osquery/config.h>
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include <boost/property_tree/xml_parser.hpp>
+#include <osquery/core/conversions.h>
+#include <osquery/core/json.h>
+#include <osquery/core/windows/wmi.h>
+
+#include "osquery/events/windows/windows_event_log.h"
+#include "osquery/filesystem/fileops.h"
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+
+FLAG(uint64,
+     windows_events_expiry,
+     60 * 60 * 24 * 30, // Keep 30 days by default
+     "Timeout to expire event subscriber results");
+
+FLAG(uint64,
+     windows_events_max,
+     100000,
+     "Maximum number of events per type to buffer");
+
+FLAG(string,
+     additional_windows_event_channels,
+     "",
+     "Additional Windows event log channels to subscribe to");
+
+class WindowsEventSubscriber
+    : public EventSubscriber<WindowsEventLogEventPublisher> {
+ public:
+  Status init() override {
+    WindowsEventLogSubscriptionContextRef wc = createSubscriptionContext();
+    for (const auto& tok :
+         osquery::split(FLAGS_additional_windows_event_channels, ",")) {
+      defaultSubscriptionChannels.insert(stringToWstring(tok));
+    }
+    wc->sources = this->defaultSubscriptionChannels;
+    subscribe(&WindowsEventSubscriber::Callback, wc);
+    return Status(0, "OK");
+  }
+
+  size_t getEventsExpiry() override {
```

Comment:
I don't think you need these, unless you really need to control these values separately.